R/MsBackend.R

```diff
@@ -27,36 +27,44 @@ NULL
 #' classes* section documents the API that a backend must implement.
 #'
 #' Currently available backends are:
-#' 
+#'
 #' - `MsBackendDataFrame`: stores all data in memory using a `DataFrame`.
-#' 
+#'
 #' - `MsBackendMzR`: stores the m/z and intensities on-disk in raw
 #'    data files (typically `mzML` or `mzXML`) and the spectra
 #'    annotation information (header) in memory in a `DataFrame`. This
 #'    backend requires the `mzR` package.
 #'
 #' See below for more details about individual backends.
 #'
-#' @param object Object extending `MsBackend`.
-#'
-#' @param x Object extending `MsBackend`.
-#'
 #' @param columns For `spectraData` accessor: optional `character` with column
 #'     names (spectra variables) that should be included in the
 #'     returned `DataFrame`. By default, all columns are returned.
 #'
+#' @param drop For `[`: not considered.
+#'
 #' @param files For `backendInitialize`: `character` with the file
 #'     names from which the data is/will be imported. Should be set to
-#'     `NA_character_` if not applicable.
+#'     `NA_character_` if not applicable (e.g. for `MsBackendDataFrame`).
 #'
 #' @param initial For `tic`: `logical(1)` whether the initially
 #'     reported total ion current should be reported, or whether the
 #'     total ion current should be (re)calculated on the actual data
 #'     (`initial = FALSE`).
 #'
+#' @param i For `[`: `integer`, `logical` or `character` to subset the object.
+#'
+#' @param j For `[`: not supported.
+#'
+#' @param object Object extending `MsBackend`.
+#'
+#' @param spectraData For : `DataFrame` with spectrum metadata/data.
+#'
 #' @param value replacement value for `<-` methods. See individual
 #'     method description or expected data type.
 #'
+#' @param x Object extending `MsBackend`.
```

Comment:
I assume that the signatures of the generics are different, but it's kind of annoying that there's `x` and `object` that refer to the same thing. And I guess it isn't possible to at least put them together using roxygen?

Reply:
Yes the signatures for the generics are different. I usually try to always use `object`, but `length` etc just have `x`. I am not aware of a way to combine them with roxygen.